docs/migration.md

```diff
@@ -3,198 +3,107 @@ title: Migration
+| "foo.+bar" | "foo\nbar"          |
+
+  If you want Prometheus v3 to behave like v2 did, you will have to change your 
+  regular expressions by replacing all `.` patterns with `[^\n]`, e.g.  
+  `foo[^\n]*`.
+- Lookback and range selectors are left open and right closed (previously left 
+  closed and right closed). This change affects queries when the evaluation time 
+  perfectly aligns with the sample timestamps. For example assume querying a 
+  timeseries with even spaced samples exactly 1 minute apart. Before Prometheus 
+  3.x, range query with `5m` will mostly return 5 samples. But if the query 
+  evaluation aligns perfectly with a scrape, it would return 6 samples. In 
+  Prometheus 3.x queries like this will always return 5 samples.
+  This change has likely few effects for everyday use, except for some sub query 
+  use cases.
+  Query front-ends that align queries usually align sub-queries to multiples of 
+  the step size. These sub queries will likely be affected.
+  Tests are more likely to affected. To fix those either adjust the expected 
+  number of samples or extend to range by less then one sample interval.
+- The `holt_winters` function has been renamed to `double_exponential_smoothing` 
+  and is now guarded by the `promql-experimental-functions` feature flag.
+  If you want to keep using holt_winters, you have to do both of these things:
+    - Rename holt_winters to double_exponential_smoothing in your queries.
+    - Pass `--enable-feature=promql-experimental-functions` in your Prometheus 
+      cli invocation..
+
+## Scrape protocols
+Prometheus v3 is more strict concerning the Content-Type header received when
+scraping. Prometheus v2 would default to the standard Prometheus text protocol
+if the target being scraped did not specify a Content-Type header or if the
+header was unparsable or unrecognised. This could lead to incorrect data being
+parsed in the scrape. Prometheus v3 will now fail the scrape in such cases.
+
+If a scrape target is not providing the correct Content-Type header the
+fallback protocol can be specified using the fallback_scrape_protocol
+parameter. See [Prometheus scrape_config documentation.](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config)
+
+This is a breaking change as scrapes that may have succeeded with Prometheus v2
+may now fail if this fallback protocol is not specified.
 
 ## Miscellaneous
 
-### Prometheus non-root user
-
-The Prometheus Docker image is now built to [run Prometheus
-as a non-root user](https://github.com/prometheus/prometheus/pull/2859). If you
-want the Prometheus UI/API to listen on a low port number (say, port 80), you'll
-need to override it. For Kubernetes, you would use the following YAML:
+### TSDB format and downgrade
+The TSDB format has been changed in Prometheus v2.55 in preparation for changes 
+to the index format. Consequently a Prometheus v3 tsdb can only be read by a 
+Prometheus v2.55 or newer.
```

Comment:
I think this needs to be super highlighted!

Before migrating to 3.0, upgrade to 2.55 and make sure everything is fine and only them move to 3.0

Reply:
(fine to be done in a separate PR)